app/controllers/spree/api/ams/orders_controller.rb

```diff
@@ -4,6 +4,15 @@ module Ams
       class OrdersController < Spree::Api::OrdersController
         include Serializable
         include Requestable
+
+        def mine
+          if current_api_user.persisted?
+            @orders = current_api_user.orders.reverse_chronological
+            respond_with(@orders)
+          else
+            render "spree/api/errors/unauthorized", status: :unauthorized
+          end
+        end
```

Comment:
This is cool - but I'd rather we detected the "ember data" signature request in the index action.  Something like this (i forget how exactly ember-data actually sends the async request):

```
def index
  if params[:ids]
    # Ensure user can load specific ids via CanCan
    # Load the collection
    # Paginate the collection if there's a per_page or page param (spree uses kaminari)
    # respond 
  else
    super
  end
end
```

The reason is that the order model is HUGE.  If a user has more than like, 2 or 3 orders, with a couple line items in each, the request will be super slow.  In those cases, I'd like it to be easy for the developer to load "some" of the orders, but not all at once - so they can paginate on the frontend.

Another consideration, is that I'd eventually like to build out a Spree Ember Backend.  Spree uses CanCan - so if the current_api_user is an admin, I'd like the index route to allow them to list all of the orders.